lib/rdstation/authentication.rb

```diff
@@ -7,9 +7,10 @@ class Authentication
     DEFAULT_HEADERS = { 'Content-Type' => 'application/json' }.freeze
     REVOKE_URL = 'https://api.rd.services/auth/revoke'.freeze
 
-    def initialize(client_id, client_secret)
-      @client_id = client_id
-      @client_secret = client_secret
+    def initialize(client_id = nil, client_secret = nil)
+      warn_deprecation if client_id || client_secret
+      @client_id = client_id || RDStation.configuration.client_id
```

Comment:
@joaohornburg foi tratado nos outros locais mas aqui faltou, o `RDStation.configuratino` pode ser `nil`, ainda que seja possível passar via argumento aqui, pode ser passado `nil` também.

Reply:
nesse caso ele vai tomar invalid credentials.

Reply:
Deveria, mas uma exception `Undefined method for nil:NilClass` estourará aqui porque se `client_id` for `nil` a expressão seguinte será executada com o `configuration` sendo `nil` .